dd-bridge-android/src/test/kotlin/com/datadog/android/bridge/internal/BridgeSdkTest.kt

```diff
@@ -907,6 +909,166 @@ internal class BridgeSdkTest {
+
+        // When
+        val proxyConfig = testedBridgeSdk.buildProxyConfiguration(config)
+
+        // Then
+        assertThat(proxyConfig).isNull()
+    }
+
+    @Test
+    fun `𝕄 not build proxy config 𝕎 buildProxyConfiguration() { port is missing }`(
+        @Forgery configuration: DdSdkConfiguration,
+        forge: Forge
+    ) {
+        // Given
+        val additionalConfig = mapOf(
+            BridgeSdk.DD_PROXY_TYPE to forge.anElementFrom("http", "https", "socks"),
+            BridgeSdk.DD_PROXY_ADDRESS to "1.1.1.1"
+        )
+        val config = configuration.copy(additionalConfig = additionalConfig)
+
+        // When
+        val proxyConfig = testedBridgeSdk.buildProxyConfiguration(config)
+
+        // Then
+        assertThat(proxyConfig).isNull()
+    }
+
+    @Test
+    fun `𝕄 build proxy configuration 𝕎 buildProxyConfiguration() { no credentials }`(
+        @Forgery configuration: DdSdkConfiguration,
+        forge: Forge
+    ) {
+        // Given
+        val proxyType = forge.anElementFrom("http", "https", "socks")
+
+        val additionalConfig = mapOf(
+            BridgeSdk.DD_PROXY_TYPE to proxyType,
+            BridgeSdk.DD_PROXY_ADDRESS to "1.1.1.1",
+            BridgeSdk.DD_PROXY_PORT to forge.anInt(min = 0, max = 65536)
+        )
+        val config = configuration.copy(additionalConfig = additionalConfig)
+
+        // When
+        val (proxy, authenticator) = testedBridgeSdk.buildProxyConfiguration(config)!!
+
+        // Then
+        assertThat(proxy.type()).matches {
+            when (proxyType) {
+                "http", "https" -> it == Proxy.Type.HTTP
+                else -> it == Proxy.Type.SOCKS
+            }
+        }
+
+        assertThat(proxy.address()).isNotNull
+
+        assertThat(authenticator).isNull()
+    }
+
+    @Test
+    fun `𝕄 build proxy configuration+authenticator 𝕎 buildProxyConfiguration() { +credentials }`(
+        @Forgery configuration: DdSdkConfiguration,
+        forge: Forge
+    ) {
+        // Given
+        val proxyType = forge.anElementFrom("http", "https", "socks")
+
+        val additionalConfig = mapOf(
+            BridgeSdk.DD_PROXY_TYPE to proxyType,
+            BridgeSdk.DD_PROXY_ADDRESS to "1.1.1.1",
+            BridgeSdk.DD_PROXY_PORT to forge.anInt(min = 0, max = 65536),
+            BridgeSdk.DD_PROXY_USERNAME to forge.anAlphabeticalString(),
+            BridgeSdk.DD_PROXY_PASSWORD to forge.anAlphabeticalString()
+        )
+        val config = configuration.copy(additionalConfig = additionalConfig)
+
+        // When
+        val (proxy, authenticator) = testedBridgeSdk.buildProxyConfiguration(config)!!
+
+        // Then
+        assertThat(proxy.type()).matches {
+            when (proxyType) {
+                "http", "https" -> it == Proxy.Type.HTTP
+                else -> it == Proxy.Type.SOCKS
+            }
+        }
+
+        assertThat(proxy.address()).isNotNull
```

Comment:
maybe you need some tests where to assert also the values from `additionalConfig` inside the `proxy` and `authenticator` instances.

Reply:
done